### DISCUSSION TITLE
POST API and Save Response To DB

### DISCUSSION THREAD

**Discussion by ikaheraJQ** [Budibase/budibase#6080] [Help] [answered]
Is anyone here know how to save POST API response to DB using one-click button?
The case is I need to get some response values from POST API and then save the values to DB in sequence.
Thanks for any kind of help 🙏🏽

**Comment by melohagan** [comment, marked as answer]
Hi @ikaheraJQ 

You could have a look at action referencing: https://docs.budibase.com/docs/referencing

So your actions would be:

- Execute Query
- Save Row `{{ Action 1.Query Result.data.0 }}`
- Save Row `{{ Action 1.Query Result.data.1 }}`
- Save Row `{{ Action 1.Query Result.data.2 }}`
- etc

This would only work if you know the number of rows returned in advance, although you can also use the **Stop If** action to check the length of the array returned, see this: https://github.com/Budibase/budibase/discussions/6070#discussioncomment-2827837

Hopefully in future there'll be looping functionality for design actions.

**Reply by ikaheraJQ** [reply]
Oh, okay I got it. Thanks for your help @melohagan 🙏🏽
